python/revng/cli/_commands/fetch_debuginfo/elf.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from pathlib import Path
+
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import NoteSection
+from xdg import xdg_data_home
+
+from .common import download_file, log
+
+
+def parse_build_id(file):
+    for section in file.iter_sections():
+        if not isinstance(section, NoteSection):
+            continue
+        for note in section.iter_notes():
+            desc = note["n_desc"]
+            if note["n_type"] != "NT_GNU_BUILD_ID":
+                continue
+            return desc
+    return None
+
+
+def fetch_dwarf(file: ELFFile, file_path, urls):
+    log("Looking for Debugging Information for an ELF")
+    build_id = parse_build_id(file)
+    if build_id is None:
+        # If we cannot parse the build id, we cannot fetch debug info.
+        log("Parsing build-id failed")
+        return None
+
+    log("BUILD-ID: " + build_id)
+
+    # Find debug info on the web as `debuginfod` does.
+    # Ensure that we have created `$xdg_data_home/revng/debug-symbols/elf` directory.
+    path_to_revng_data = Path(str(xdg_data_home()) + "/revng/")
```

Comment:
`xdg_cache_home` would probably be more appropriate. What do you think?

Reply:
We have used ```xdg_cache_home``` at first, and decided to move to ```xdg_data_home``` :) But I guess that ```cache``` is more appropriate.

Reply:
If it's something we can fetch/recreate again (unless the apocalypse ensues), let's go for cache.